src/signals-mach.c

```diff
@@ -167,11 +161,11 @@ static void jl_call_in_state(jl_ptls_t ptls2, host_thread_state_t *state,
     uint64_t rsp = (uint64_t)ptls2->signal_stack + sig_stack_size;
     assert(rsp % 16 == 0);
 
+#ifdef _CPU_X86_64_
```

Comment:
I meant to put this in a separate PR - oh well. It's correct anyway, and I'll follow up with the lr reset in a follow up PR.